backend/cfg/cfg_with_layout.ml

```diff
@@ -173,3 +173,44 @@ let save_as_dot t ?show_instr ?show_exn ?annotate_block ?annotate_succ msg =
       print_dot ?show_instr ?show_exn ?annotate_block ?annotate_succ ppf t)
     ~always:(fun () -> close_out oc)
     ~exceptionally:(fun _exn -> Misc.remove_file filename)
+
+module Permute = struct
+  (* Implementation of this module is copied from Base *)
+  external unsafe_set : 'a array -> int -> 'a -> unit = "%array_unsafe_set"
+
+  let swap t i j =
+    let elt_i = t.(i) in
```

Comment:
(Provided we give a signature to `Permute`
in order to hide `swap`, we could as well
use unsafe gets.)

Reply:
Right, we should consider it for Base. Let's leave it unoptimized here, it'll defend against bugs in `Permute.array`.

Reply:
Well, no, in Base `swap` is accessible to the user,
so the bounds should be checked. My point was
that in this pull request we control when `swap`
is called.

Reply:
I forgot that in Base swap is accessible.